networkx/algorithms/dag.py

```diff
@@ -174,39 +249,9 @@ def topological_sort(G, with_generation=False):
     .. [1] Manber, U. (1989).
        *Introduction to Algorithms - A Creative Approach.* Addison-Wesley.
     """
-    if not G.is_directed():
-        raise nx.NetworkXError("Topological sort not defined on undirected graphs.")
-
-    indegree_map = {v: d for v, d in G.in_degree() if d > 0}
-    # These nodes have zero indegree and ready to be returned.
-    zero_indegree = deque(v for v, d in G.in_degree() if d == 0)
-
-    generation = 0
-    last_node_in_generation = zero_indegree[-1] if zero_indegree else None
-
-    while zero_indegree:
-        node = zero_indegree.popleft()
-        if node not in G:
-            raise RuntimeError("Graph changed during iteration")
-        for _, child in G.edges(node):
-            try:
-                indegree_map[child] -= 1
-            except KeyError as e:
-                raise RuntimeError("Graph changed during iteration") from e
-            if indegree_map[child] == 0:
-                zero_indegree.append(child)
-                del indegree_map[child]
-
-        yield (node, generation) if with_generation else node
-
-        if zero_indegree and node == last_node_in_generation:
-            generation += 1
-            last_node_in_generation = zero_indegree[-1]
-
-    if indegree_map:
-        raise nx.NetworkXUnfeasible(
-            "Graph contains a cycle or graph changed " "during iteration"
-        )
+    for generation in nx.topological_generations(G):
+        for node in generation:
+            yield node
```

Comment:
```suggestion
        yield from generation
```
I **think** this is a case where `yield from` makes sure the exception handling etc are done well.
Even if it isn't, it is probably good practice to use `yield from` in cases like this where you are just yielding from a generator.